File: C#/0107. Binary Tree Level Order Traversal II.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrderBottom(TreeNode root) {
        IList<IList<int>> rep= new List<IList<int>>();
        if (root==null){
            return rep;
        }
        
        Queue<TreeNode> rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        while(rootQueue.Count!=0){
            IList<int> cur=new List<int>();
            int lenRootQueue=rootQueue.Count;
            for(int i=0;i<lenRootQueue;i++){
                TreeNode curRoot=rootQueue.Dequeue();
                cur.Add(curRoot.val);
                if (curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                }
                if (curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                }
            }
            rep.Insert(0,cur);
        }
        return rep;
    }
}

File: 20200215Python-China/0107. Binary Tree Level Order Traversal II.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        rep=[]
        root_list=[root]
        while root_list:
            len_of_root_list=len(root_list)
            cur_list=[]
            for i in range(len_of_root_list):
                cur_root=root_list.pop(0)
                cur_list.append(cur_root.val)
                if cur_root.left:
                    root_list.append(cur_root.left)
                if cur_root.right:
                    root_list.append(cur_root.right)
            rep.insert(0,cur_list)
        return rep
        
